src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java

```diff
@@ -651,9 +662,13 @@ private int storePartialItemStack(final ItemStack itemStackIn)
      */
     private int storeItemStack(final ItemStack itemStackIn)
     {
-        if (this.canMergeStacks(this.getStackInSlot(this.currentItem), itemStackIn))
+        if (this.canMergeStacks(this.getStackInSlot(this.mainItem), itemStackIn))
+        {
+            return this.mainItem;
+        }
+        else if (this.canMergeStacks(this.getStackInSlot(this.offhandItem), itemStackIn))
         {
-            return this.currentItem;
+            return this.offhandItem;
         }
         else if (this.canMergeStacks(this.getStackInSlot(40), itemStackIn))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 40 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)